docs/library-usage.md

```diff
@@ -0,0 +1,25 @@
+# Cloud SQL proxy dialer for Go
+
+You can also use the Cloud SQL proxy directly from a Go program.
+
+These packages are primarily used as implementation for the Cloud SQL proxy
+command, and may be changed in backwards incompatible ways in the future.
+
+## To use inside a Go program:
+
+If your program is written in [Go](https://golang.org) you can use the Cloud SQL
+Proxy as a library, avoiding the need to start the Proxy as a companion process.
+
+### MySQL
```

Comment:
Can you add a section here for using the Dial function directly? See #433

Reply:
Given that we're working on structural improvements to the proxy, should we wait on this since its API will likely change?